modules/prebid/rulesengine/module.go

```diff
@@ -104,19 +109,31 @@ func (m Module) Shutdown() {
 }
 
 // rebuildTrees returns true if the trees for this account need to be rebuilt; false otherwise
-func rebuildTrees(co *cacheEntry, jsonConfig *json.RawMessage) bool {
-	if !expired(&timeutil.RealTime{}, co.timestamp) {
-		return false
+func rebuildTrees(co *cacheEntry, jsonConfig *json.RawMessage) (bool, error) {
+	updateFrequency, err := jsonparser.GetInt(*jsonConfig, "updatetreefrequencyminutes")
```

Comment:
Consider moving the reading of the variable to the init / builder function, which returns this to a simpler form without the error return.

That would mean this value can never change after start up, however the alternative of parsing the json on every request (that's what's happening here, right?) is very expensive. Not clear on what a compromise would look like.

Reply:
Discussed offline. This logic will be moved to the Cache init.

Reply:
Addressed

Reply:
For other reviewers, this means if the refresh rate is 0 (do not refresh) at app startup, the refresh timer will forever be 0 unless the app is restarted with a different value. Happy to consider alternatives which still allow for a setting change, which also doesn't check config for each auction (that's too computationally expensive).